### ISSUE TITLE
Problem reproducing MultiRoom-N10(N12)-SX

### ISSUE THREAD

**Issue by Mathieu-Seurin** [facebookresearch/impact-driven-exploration#4]
Hello,

Thanks for sharing the code for this paper, we greatly appreciated your implementations!

I'm running into a problem trying to reproduce MultiRoom-N12-S10 (and MultiRoom-N10-S10/S4)
RIDE agent is not able to learn in those environments using the hyperparameter given in the paper.

Entropy_coef : 0.001 
Intrinsic reward coef : 0.5 
(for N12-S10)

It's able to reach the last room sometimes but not consistently. The final score remains close to 0
Was it run on the procedurally-generated version? From the paper, it seems so, but can it be that it was run on a singleton environment?

Are there any other hyperparameters that I forgot or that were not indicated in the paper? (Maybe weight of losses etc..)

Thanks in advance

**Comment by rraileanu** [comment]
Hi! This is surprising to hear. I know others were able to reproduce the results independently. 

Is the total_frames set to 1e8? The code is using a learning rate scheduler so if you are setting the total_frames to a lower number it might change the training dynamics and end up with too low a learning rate to train effectively. 

Other the entropy_coef and intrinsic_reward_coef, the hyperparameter values should be the default ones in arguments.py. The important ones are: num_actors=40, num_buffers=80, unroll_length=100, num_threads=4, batch_size=32, learning_rate=0.0001, forwards_dynamics_loss=10.0, inverse_dynamics_loss=0.1, so make sure you are using these values. I also remember obtaining similar results with entropy_coef=0.0007 and 0.0005 but 0.001 should work too. 

And yes, we definitely ran it in procedurally-generated mode. 

Please let me know if you're still having problems reproducing those results.

**Comment by Mathieu-Seurin** [comment]
It might be the scheduler indeed, as I ran the experiments using a smaller amount of frame
The others parameters seem fine to me, I'll try using 1e8 frames !

I'll keep you updated, thanks

**Comment by Mathieu-Seurin** [comment]
Hi, by setting the total frame to 1e8, I was able to reproduce your results on MultiRoom-N12-S10
Thanks !

**Comment by CrazySssst** [comment]
> Hi, by setting the total frame to 1e8, I was able to reproduce your results on MultiRoom-N12-S10
> Thanks !

Hi, i try to reproduce results on many Mini-Grid environments, but I always get 0 rewards.  Could give me some advices?
